templates/workloads/partials/cf/publish.yml

```diff
@@ -0,0 +1,31 @@
+{{- if .Publish}}
+{{- range $topic := .Publish.Topics}}
+{{.Name}}SNSTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: !Join ['', [!Ref AppName, '-', !Ref EnvName, '-', {{.Name}}]]
+      
+  {{.Name}}SNSTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    DependsOn: {{.Name}}SNSTopic
+    Properties:
+      Topics: 
+        - !Ref ${AWS::StackName}-{{.Name}}-SNSTopic
```

Comment:
Why not 
```suggestion
        - !Ref {{.Name}}SNSTopic
```
instead to refer to [this resource](https://github.com/aws/copilot-cli/pull/2586/files#diff-12636081a605ca709021df639febb16ea5b1bc8297f9f6fd24e03193f22902aeR3)?

Reply:
Adjusted - this got mixed up with the name.